### PR TITLE
docs: ADR 011

### PR DESCRIPTION
**Pull request by opqdonut**

### CODE REVIEW

File: docs/architecture/011-api-key-access-control.md [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+Current approach: A new `:application-closer` role, and an API key
+associated with it. Also needs fixes to the handling of implicit
+roles.
+
+Proposed approach: An API key associated with a user that is a handler
+for suitable workflows. The API key is limited to the
+`/api/applications/close` POST request, and some additional GET
+requests as needed.
+
```
Comment (opqdonut) [inline docs/architecture/011-api-key-access-control.md:87]: more use cases?

File: docs/architecture/011-api-key-access-control.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+_implicit roles_ that are granted per application, based on the
+application state. For example, a user has the `:reviewer` role for an
+application if they have been invited to review the application. When
+the API key roles were implemented, they were (accidentally?) only
+implemented for explicit roles. Extending the implementation for
+implicit roles would require more code.
+
+As a concrete example, an API key with only the role `:logged-in`
+would still be able to impersonate the handler of an application and
+perform handler commands like approving the applicaiton.
```
Comment (Macroz) [inline docs/architecture/011-api-key-access-control.md:30]: > application.

Comment (Macroz) [reply docs/architecture/011-api-key-access-control.md:30]: How about add here the mention of the background why the previous system was implemented, e.g. the requirements from NKR which will be elaborated in the use cases? That's why we also consider this change now.

File: docs/architecture/011-api-key-access-control.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+roles. Additional work would be required to actually support the
+`:applicant` role for an API key (it's an implicit role).
+
+Proposed approach: Two API keys:
+
+- One API key for creating users. It can only impersonate a fixed user
+  with the `:user-owner` role. (Can also be limited to the
+  `/api/users/create` endpoint.)
+- Another API key for creating applications. It can impersonate
+  anyone, but is limited to the `/api/application/save-draft` and
```
Comment (Macroz) [inline docs/architecture/011-api-key-access-control.md:63]: Also `/api/applications/create`?

File: docs/architecture/011-api-key-access-control.md (hunk 2 above)
Comment (opqdonut) [reply docs/architecture/011-api-key-access-control.md:30]: Typo fixed, paragraph added.

File: docs/architecture/011-api-key-access-control.md (hunk 3 above)
Comment (opqdonut) [reply docs/architecture/011-api-key-access-control.md:63]: done

**Pull request merged**